### PR TITLE
ci: Cleanup Makefile and setup scripts

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR will remove not used targets in Makefile and not used functions in setup_dev.

`rustup` will install toolchains correctly, we don't need to care about them.

### CODE REVIEW

File: scripts/setup/dev_setup.sh [hunk 1]
```diff
@@ -296,17 +296,6 @@ function install_cargo_binary {
 	fi
 }
 
-function install_toolchain {
-	version=$1
-	echo "==> Installing ${version} of rust toolchain..."
-	rustup install "$version"
```
Comment (everpcpc) [inline scripts/setup/dev_setup.sh:302]: this step is needed for build-tool image

Comment (Xuanwo) [reply scripts/setup/dev_setup.sh:302]: Addressed by running `cargo version`

Comment (Xuanwo) [reply scripts/setup/dev_setup.sh:302]: For more background: https://note.xuanwo.io/#/page/Rustup

Comment (everpcpc) [reply scripts/setup/dev_setup.sh:302]: this is needed for cross build, since it would only install x86 gnu

Comment (everpcpc) [reply scripts/setup/dev_setup.sh:302]: Oh my mistake, cross toolchain setup is in separate command.

**Review by everpcpc — APPROVED** [review]

File: scripts/setup/dev_setup.sh (hunk 1 above)
Comment (everpcpc) [reply scripts/setup/dev_setup.sh:302]: so  `cargo version` is not needed.

**Review by PsiACE — APPROVED** [review]

**Pull request merged**